### PR TITLE
feat: introduce value-align for text inputs

### PR DESCRIPTION
**Pull request by dlockhart**
Following up from #3064, this adds a `value-align` attribute to text and number inputs. This removes the need for the `--d2l-input-text-align` CSS variable, and by having it be an attribute we can now move the validation icon to the opposite side of the value text.

Screen Shot 2022-12-09 at 10 00 03 AM

### CODE REVIEW

File: components/inputs/input-number.js [hunk 1]
```diff
@@ -331,6 +337,7 @@ class InputNumber extends FocusMixin(LabelledMixin(SkeletonMixin(FormElementMixi
 	}
 
 	render() {
+		const valueAlign = (this.valueAlign === 'end') ? 'end' : 'start';
```
Comment (dlockhart) [inline components/inputs/input-number.js:340]: The validation stuff hates it if you just assign an arbitrary value to an enum field. This seems to help it.

**Review by tdevos-d2l — APPROVED** [review]
That was quick, thank you! :shipit:

**Comment by tdevos-d2l** [comment]
Something that I forgot to bring up in the last PR, should the unit have a different colour than the editable text? Makes the user think they should be able to modify the unit.

File: components/inputs/input-text.js [hunk 2]
```diff
@@ -401,8 +410,8 @@ class InputText extends FocusMixin(LabelledMixin(FormElementMixin(SkeletonMixin(
 		const lastSlotName = (this.dir === 'rtl') ? 'left' : 'right';
 
 		const isValid = ariaInvalid !== 'true' || this.disabled;
-		const invalidIconSide = (this.dir === 'rtl') ? 'left' : 'right';
-		const invalidIconOffset = Math.max((this.dir === 'rtl') ? this._firstSlotWidth : this._lastSlotWidth, 12);
+		const invalidIconSide = ((this.dir === 'rtl' && this.valueAlign === 'start') || (this.dir !== 'rtl' && this.valueAlign === 'end')) ? 'left' : 'right';
+		const invalidIconOffset = Math.max((invalidIconSide === 'left') ? this._firstSlotWidth : this._lastSlotWidth, 12);
```
Comment (dlockhart) [inline components/inputs/input-text.js:414]: All this logic would be a lot simpler I think if we used logical CSS properties for everything, but I wasn't quite ready for that scale of change.

**Review by dbatiste — APPROVED** [review]
Looks good!

**Comment by dbatiste** [comment]
I think there are cases (ex. -1002 below) where it's not obvious the number is being hidden until they focus, but it is possible to produce the same sorta thing with the value aligned to the start as well.

**Comment by tdevos-d2l** [comment]
Is the position of the text that's input being recalculated when the `unit` changes? I'm going from `unit='/9'` to `unit='/1112'`
I know it's not common for the unit to be changed, but in our (CE) use case it *can* happen 😬 

https://user-images.githubusercontent.com/90270112/206801572-a3ca34d6-2fd4-47db-a4eb-4f2655dd20dd.mp4

**Comment by dlockhart** [comment]
> Is the position of the text that's input being recalculated when the unit changes?

It isn't, no -- good catch. I'll follow up with a bug fix PR.

**Pull request merged**